compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/hapticfeedback/HapticFeedback.uikit.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.hapticfeedback
+
+import platform.UIKit.UIImpactFeedbackGenerator
+import platform.UIKit.UISelectionFeedbackGenerator
+
+// TODO: minor UX improvement, add `prepare()` calls when internal APIs that use HapticFeedback
+//  to reduce haptic feedback latency
+//  see https://developer.apple.com/documentation/uikit/uifeedbackgenerator
+internal fun HapticFeedback() = object : HapticFeedback {
```

Comment:
Any reason to have `fun` + anonymous `object` instead of just a `class`?

Reply:
Not really, looked liked a fine place to use this idiom, but made it `CupertionHapticFeedback` class